ocaml/xapi/binpack.ml

```diff
@@ -171,6 +172,24 @@ let simulate_failure config dead_host =
   let placement = List.map (fun (vm, host) -> vm, (if List.mem_assoc vm plan then assoc "simulate_failure" vm plan else host)) config.placement in
   { config with hosts = hosts; placement = placement; num_failures = config.num_failures - 1 }
 
+let simulate_failure_approximation config =
+  let rec sum_of_first_n n l = match l with
```

Comment:
I am not convinced by this function. I believe it is better to combine these functions:

* `sum: int list -> int`
* `take: int -> 'a list -> 'a list` (`Xapi_stdext_std__Listext.List.take`)
* `drop: int -> 'a list -> 'a list` (drops first n elements)